From 2b0803f7a27ddb49fe8c7a425113a0cc030c3928 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 25 Jan 2021 14:13:22 +0700
Subject: [PATCH 424/433] ARM: dts: r8a7742-iwg21d-q7-dbcm-ca: Disable QSPI to
 enable VIN2

On the iWave RZ/G1H, QSPI and VIN2 are multiplexed. This patch hogs
select GPIO GP0_18 to HIGH and mark QSPI node status as disabled.
By doing so, VIN2 is chosen to enable.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts
index 5e34d0e..a3ac85e 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts
@@ -46,6 +46,19 @@
 	};
 };
 
+&gpio0 {
+	/* disable hogging GP0_18 to output LOW */
+	/delete-node/ qspi_en;
+
+	/* hog GP0_18 to output HIGH  to enable VIN2 */
+	qspi_en {
+		gpio-hog;
+		gpios = <18 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "VIN2_EN";
+	};
+};
+
 &hscif0 {
 	pinctrl-0 = <&hscif0_pins>;
 	pinctrl-names = "default";
@@ -248,6 +261,11 @@
 	status = "okay";
 };
 
+&qspi {
+	/* Pins shared with VIN2, keep status disabled */
+	status = "disabled";
+};
+
 &scif0 {
 	pinctrl-0 = <&scif0_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

